tests/testthat/test-weekly.R

```diff
@@ -0,0 +1,643 @@
+  ## test that the weekly R estimates from weekly data match exactly the weekly
+  ## R estimates from the daily reconstructed incidence
+  
+  res_daily_reconstructed <- suppressWarnings(estimate_R(incid = res_weekly$I, 
+                                        config = config,
+                                        method = method))
+  
+  common_t_start <- intersect(res_daily_reconstructed$R$t_start, res_weekly$R$t_start)
+  
+  relative_error <- abs(res_daily_reconstructed$R$`Mean(R)`[res_daily_reconstructed$R$t_start %in% common_t_start] -   
+                          res_weekly$R$`Mean(R)`[res_weekly$R$t_start %in% common_t_start]) / 
+    res_daily_reconstructed$R$`Mean(R)`[res_daily_reconstructed$R$t_start %in% common_t_start]
+  
+  expect_true(all(relative_error < 1e-9)) 
+  
+  ######################################################################
+  ## test that the weekly R estimates from estimate_R are the same as from 
+  ## estimate_R_agg
+  
+  expect_true(all(res_weekly$R == res_weekly_agg$R)) 
+  
+})
+
+
+
+test_that("r grid can be automatically updated with similar results", {
+  ## estimate the reproduction number (method "parametric_si")
+  ## when not specifying t_start and t_end in config, they are set to estimate
+  ## the reproduction number on sliding weekly windows      
+  method <- "parametric_si"
+  config <- make_config(list(mean_si = mean_si,
+                             std_si = std_si))
+  
+ res_weekly_small_grid <- suppressWarnings(estimate_R_agg(incid = weekly_inc, 
+                               dt = 7L, 
+                               dt_out = 7L, 
+                               iter = 10L,
+                               config = config,
+                               method = method,
+                               grid = list(precision = 0.001, min = 0.01, max = 0.012)))
+ 
+ res_weekly_default_grid <- suppressWarnings(estimate_R_agg(incid = weekly_inc, 
+                                         dt = 7L, 
+                                         dt_out = 7L, 
+                                         iter = 10L,
+                                         config = config,
+                                         method = method))
+ 
+ expect_true(max(abs(res_weekly_small_grid$R - res_weekly_default_grid$R)) < 1e-9)
+  
+})
+
+
+
+test_that("dt and dt_out in estimate_R_agg are in the correct format", {
+  method <- "parametric_si"
+  config <- make_config(list(mean_si = mean_si,
+                             std_si = std_si))
+  
+  expect_error(suppressWarnings(estimate_R_agg(incid = weekly_inc, 
+                                               dt = 7, 
+                                               dt_out = 7L, 
+                                               iter = 10L,
+                                               config = config,
+                                               method = method)),
+               "dt and dt_out must be integers e.g. 7L") 
```

Comment:
that's not technically true i.e. dt can be a vector of integers so maybe update the error message and update in tests too